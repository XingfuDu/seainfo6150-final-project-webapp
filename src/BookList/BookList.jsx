import React from 'react';
import PropTypes from 'prop-types';
import classes from './BookList.module.css';
import BookListItem from '../BookListItem/BookListItem';

const BookList = (props) => {
	let displayContent;

	if (props.books.length) {
		displayContent = (
			<div>
				<div className={classes.readingChallengeBanner}>
					<div className={classes.bannerYear} style={{"backgroundColor": "#f2e9e4"}}>
						<img
							alt="2021 Reading Challenge"
							className={classes.bookImg}
							src="https://s.gr-assets.com/assets/challenges/yearly/img_RCBook-626ef5100b911207af0c4d84f02df03a.svg"
						/>
						<div className={classes.year}>
							<span>
								2021 READING
								<br />
							</span>
							<span>CHALLENGE</span>
						</div>
					</div>
					<div className={classes.challengeStatsContainer2}>
						<table>
							<tbody>
								<tr>
									<td className={classes.challengeStatItem}>Participants</td>
									<td className={classes.statNumber}>3,711,771</td>
								</tr>
								<tr>
									<td className={classes.challengeStatItem}>Books Pledged</td>
									<td className={classes.statNumber}>181,599,755</td>
								</tr>
								<tr>
									<td className={classes.challengeStatItem}>Avg. Books Pledged</td>
									<td className={classes.statNumber}>47</td>
								</tr>
								<tr>
									<td className={classes.challengeStatItem}>Time Left</td>
									<td className={classes.statNumber}>253 days, 13 hours</td>
								</tr>
							</tbody>
						</table>
					</div>
				</div>
				<ul className={classes.container}>
					{props.books.map((book) => <BookListItem book={book} key={book.id} />)}
				</ul>
			</div>
		);
	} else {
		displayContent = <div className={classes.noData}>You have no data!</div>;
	}

	return displayContent;
};

BookList.propTypes = {
	books: PropTypes.array.isRequired
};
export default BookList;
