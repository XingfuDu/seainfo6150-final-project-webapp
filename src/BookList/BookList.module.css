.container {
	background: white;
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	gap: 20px;
	font-family: 'Lora', serif;
	list-style-type: none;
	padding: 15px;
	min-height: 100%;
	width: 100%;
	box-sizing: border-box;
}

.noData {
	display: grid;
	align-items: center;
	justify-content: center;
	font-size: 150%;
	color: #3d5a80;
	min-height: 100%;
	width: 100%;
}

@media (max-width: 1200px) {
	.container {
		padding: 18px;
		grid-template-columns: repeat(2, 1fr);
	}
}

@media (max-width: 800px) {
	.container {
		grid-template-columns: repeat(1, 1fr);
		padding: 20px;
	}
}

@media (max-width: 400px) {
	.container {
		padding: 22px;
	}
}


