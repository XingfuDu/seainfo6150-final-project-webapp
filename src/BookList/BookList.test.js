import React from 'react';
import BookList from './BookList';
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';

describe('BookList tests', () => {
    const data = {  
        "0": {
        "id": "0",
        "categoryId": "1",
        "category": "Psychology",
        "title": "Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals",
        "author": "Hal Herzog",
        "description": "<p>Combining the intellect of Malcolm Gladwell with the irreverent humor of Mary Roach and the paradigm-shifting analysis of Jared Diamond, a leading social scientist offers an unprecedented look inside our complex and often paradoxical relationships with animals.</p><p>Does living with a pet really make people happier and healthier? What can we learn from biomedical research with mice? Who enjoyed a better quality of life—the chicken on a dinner plate or the rooster who died in a Saturday-night cockfight? Why is it wrong to eat the family dog? Drawing on more than two decades of research in the emerging field of anthrozoology, the science of human–animal relations, Hal Herzog offers surprising answers to these and other questions related to the moral conundrums we face day in and day out regarding the creatures with whom we share our world.</p><p>Some We Love, Some We Hate, Some We Eat is a highly entertaining and illuminating journey through the full spectrum of human–animal relations, based on Dr. Herzog’s groundbreaking research on animal rights activists, cockfighters, professional dog-show handlers, veterinary students, and biomedical researchers. Blending anthropology, behavioral economics, evolutionary psychology, and philosophy, Herzog carefully crafts a seamless narrative enriched with real-life anecdotes, scientific research, and his own sense of moral ambivalence.</p><p>Alternately poignant, challenging, and laugh-out-loud funny, this enlightening and provocative book will forever change the way we look at our relationships with other creatures and, ultimately, how we see ourselves.",
        "image": "https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1369453733l/6953508.jpg",
        "price": "12.99",
        "timeStamp": "2018-11-22T17:55:57.000Z",
        "displayDate": "November 22nd 2018, 9:55 am"
      },
      "1": {
        "id": "1",
        "categoryId": "1",
        "category": "Psychology",
        "title": "Remember: The Science of Memory and the Art of Forgetting",
        "author": "Lisa Genova",
        "description": "<p>A fascinating exploration of the intricacies of how we remember, why we forget, and what we can do to protect our memories, from the Harvard-trained neuroscientist and bestselling author of Still Alice.</p><p>Have you ever felt a crushing wave of panic when you can't for the life of you remember the name of that actor in the movie you saw last week, or you walk into a room only to forget why you went there in the first place? If you're over forty, you're probably not laughing. You might even be worried that these lapses in memory could be an early sign of Alzheimer's or dementia. In reality, for the vast majority of us, these examples of forgetting are completely normal. Why? Because while memory is amazing, it is far from perfect. Our brains aren't designed to remember every name we hear, plan we make, or day we experience. Just because your memory sometimes fails doesn't mean it's broken or succumbing to disease. Forgetting is actually part of being human.</p><p>In Remember, neuroscientist and acclaimed novelist Lisa Genova delves into how memories are made and how we retrieve them. You'll learn whether forgotten memories are temporarily inaccessible or erased forever and why some memories are built to exist for only a few seconds (like a passcode) while others can last a lifetime (your wedding day). You'll come to appreciate the clear distinction between normal forgetting (where you parked your car) and forgetting due to Alzheimer's (that you own a car). And you'll see how memory is profoundly impacted by meaning, emotion, sleep, stress, and context. Once you understand the language of memory and how it functions, its incredible strengths and maddening weaknesses, its natural vulnerabilities and potential superpowers, you can both vastly improve your ability to remember and feel less rattled when you inevitably forget. You can set educated expectations for your memory, and in doing so, create a better relationship with it. You don't have to fear it anymore. And that can be life-changing.",
        "image": "https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1598831596l/54895704.jpg",
        "price": "13.52",
        "timeStamp": "2018-11-22T15:12:24.000Z",
        "displayDate": "November 22nd 2018, 7:12 am"
      }
    };
    const books = Object.values(data);
	it('renders correctly', () => {
		const { container } = render(
			<BrowserRouter><BookList books={books}/></BrowserRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
