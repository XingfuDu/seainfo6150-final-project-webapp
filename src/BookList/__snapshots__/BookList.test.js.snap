// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="readingChallengeBanner"
    >
      <div
        class="bannerYear"
        style="background-color: rgb(242, 233, 228);"
      >
        <img
          alt="2021 Reading Challenge"
          class="bookImg"
          src="https://s.gr-assets.com/assets/challenges/yearly/img_RCBook-626ef5100b911207af0c4d84f02df03a.svg"
        />
        <div
          class="year"
        >
          <span>
            2021 READING
            <br />
          </span>
          <span>
            CHALLENGE
          </span>
        </div>
      </div>
      <div
        class="challengeStatsContainer2"
      >
        <table>
          <tbody>
            <tr>
              <td
                class="challengeStatItem"
              >
                Participants
              </td>
              <td
                class="statNumber"
              >
                3,711,771
              </td>
            </tr>
            <tr>
              <td
                class="challengeStatItem"
              >
                Books Pledged
              </td>
              <td
                class="statNumber"
              >
                181,599,755
              </td>
            </tr>
            <tr>
              <td
                class="challengeStatItem"
              >
                Avg. Books Pledged
              </td>
              <td
                class="statNumber"
              >
                47
              </td>
            </tr>
            <tr>
              <td
                class="challengeStatItem"
              >
                Time Left
              </td>
              <td
                class="statNumber"
              >
                253 days, 13 hours
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <img
          alt="Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals"
          class="image"
          src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1369453733l/6953508.jpg"
        />
        <div
          class="wrapper"
        >
          <h3>
            <a
              class="link"
              href="/book/1/0"
            >
              Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals
            </a>
          </h3>
          <time
            class="time"
            datetime="2018-11-22T17:55:57.000Z"
          >
            November 22nd 2018, 9:55 am
          </time>
          <p
            class="author"
          >
            By 
            Hal Herzog
          </p>
        </div>
      </li>
      <li
        class="container"
      >
        <img
          alt="Remember: The Science of Memory and the Art of Forgetting"
          class="image"
          src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1598831596l/54895704.jpg"
        />
        <div
          class="wrapper"
        >
          <h3>
            <a
              class="link"
              href="/book/1/1"
            >
              Remember: The Science of Memory and the Art of Forgetting
            </a>
          </h3>
          <time
            class="time"
            datetime="2018-11-22T15:12:24.000Z"
          >
            November 22nd 2018, 7:12 am
          </time>
          <p
            class="author"
          >
            By 
            Lisa Genova
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
